### ISSUE TITLE
BUG? micromed io

### ISSUE THREAD

**Issue by dengemann** [NeuralEnsemble/python-neo#360]
Hi neighbors,

we're taking another look at reading micromed data into MNE

https://github.com/mne-tools/mne-python/issues/4359
https://github.com/mne-tools/mne-python/issues/1605

And I just realized that the micromed io seems to be broken in Python 3:
https://github.com/NeuralEnsemble/python-neo/blob/master/neo/io/micromedio.py#L79

Here the file object is replaced by a BufferedReader which expects apparently an integer as input.

Trying to use example code:
```Python
filename =  'EEG_1.TRC'
seg = neo.io.MicromedIO(filename=filename)
seg.read_segment(lazy=False, cascade=True)
```

gives ugly errors:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-39-e6a3cafb8ed5> in <module>()
      5     seg = neo.io.MicromedIO(filename=filename)
      6     seg.filename
----> 7     seg.read_segment(lazy=False, cascade=True)

/Users/dengemann/anaconda/lib/python3.5/site-packages/neo/io/micromedio.py in read_segment(self, cascade, lazy)
     77         Arguments:
     78         """
---> 79         f = StructFile(self.filename, 'rb')
     80 
     81         # Name

TypeError: 'str' object cannot be interpreted as an integer
```

Any thoughts?

**Comment by dengemann** [comment]
I confirm that it does not crash in Python 2.7

**Comment by samuelgarcia** [comment]
This is anoying.
I see in neo.tests.iotest.test_micromedio that it is not py3 compatible!!
We need to fix it.

**Comment by samuelgarcia** [comment]
The Io is very old it should be written entirely...
I made a quite and dirty fix here PR #361.

Could you test ?

**Comment by dengemann** [comment]
Thanks! I will have a look in a moment.

**Comment by johntukker** [comment]
I have a similar issue with neuroexplorerio.py, don't know if I should open this as a new issue but seems to be same so would be very happy if someone could have a look (I'm new to Python myself, so sorry if this is an obvious fix but even after looking at PR#361 still can't figure it out). This is running Python 3.4.5 with Jupyter.

r = io.NeuroExplorerIO(filename)
seg = r.read_segment(lazy=False, cascade=True)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-29-f75e95c4a5b3> in <module>()
      1 r = io.NeuroExplorerIO(filename)
----> 2 seg = r.read_segment(lazy=False, cascade=True)

C:\Users\tukkerjj\miniconda3\lib\site-packages\neo\io\neuroexplorerio.py in read_segment(self, lazy, cascade)
     89         for i in range(global_header['nvar']):
     90             entity_header = HeaderReader(fid, EntityHeader).read_f(
---> 91                 offset=offset + i * 208)
     92 
     93             print 'i',i, entityHeader['type']

TypeError: 'str' does not support the buffer interface

cheers,
John
